text_generator.go

```diff
@@ -103,51 +122,44 @@ func (p *textPlistGenerator) writeIndent() {
 	}
 }
 
-/*
-	writePlistValue(pval cfValue, meta Meta)
-	This would need to take in a meta value and enrich the node.
+func (p *textPlistGenerator) plistAnnotation(str string) cfAnnotation {
+	return cfAnnotation{value:str}
+}
 
-*/
-func (p *textPlistGenerator) writePlistValue(pval cfValue) {
+func (p *textPlistGenerator) writePlistValue(pval cfValue, nodes ...Node) {
 	if pval == nil {
 		return
 	}
+	switch pval := pval.(type) {
+	case *cfDictionary:
+		var keys []string
+		var values []cfValue
+		if len(nodes) > 0 {
+			keys, values, nodes = pval.filterNodes(nodes)
```

Comment:
I think that if `nodes` is `nil` the `filterNodes` code will panic due to a nil pointer reference